packages/unified-selection/README.md

```diff
@@ -132,3 +132,35 @@ const selectionHiliteSet = await selectionHiliteProvider.getHiliteSet({ imodel.k
 // is shorter than that of `SelectionStorage`, to unregister the listener.
 selectionHiliteProvider.dispose();
 ```
+
+#### Selection scopes
+
+Selection scopes allow decoupling of what gets picked and what gets selected. Without selection scopes, whenever a user picks an element in the viewport, its ID goes straight into unified selection storage. With selection scopes we can modify that and add something different. The input to selection scopes' processor is a query executor, element IDs and the scope to apply, and the output an iterator of `SelectableInstanceKey`. We get the input when user picks some elements in the viewport, run that through selection scope processor and put the output into unified selection storage.
+
+Here are the scopes we support at the moment:
+
+- `element` - return key of selected element
+- `category` - return key of element's category
+- `model` - return key of element's model
+- `functional` - return key of element's related functional element
+
+The `@itwin/unified-selection` package delivers a `computeSelection` function for computing which elements should be added into unified selection storage based on the selected element ID's and a specified selection scope:
+
+```ts
+import { computeSelection } from "@itwin/unified-selection";
+import { IModelConnection } from "@itwin/core-frontend";
+import { createECSqlQueryExecutor } from "@itwin/presentation-core-interop";
+const queryExecutor = createECSqlQueryExecutor(imodel);
+const selection = computeSelection({ queryExecutor, elementIds, scope: "element" });
+```
+
+`element` and `functional` scopes additionally allow selecting assembly elements by specifying the `ancestorLevel` property in the selection scope argument of `computeSelection` function. The `ancestorLevel` property specifies how far "up" we should walk to find the target element. When not specified or `0`, the target element matches the request element. When set to `1`, the target element matches the direct parent element. When `2`, the target element is parent of the parent element and so on. In all situations when this is `> 0`, we're not walking further than the last existing element, for example when `ancestorLevel = 1` (direct parent element is requested), but the request element doesn't have a parent, the request element is returned as the result. A negative value would result in the top-most element to be returned.
```

Comment:
I think this needs to additionally mention how ancestors are handled for functional elements - do we walk "up" the input elements or their related functional elements.

Reply:
Added another paragraph for the functional elements.